projects/graph/src/graph.py

```diff
@@ -71,25 +71,30 @@ def bfs(self, starting_vertex, target):  # Breath-First Search
                     return current_path
                 else:
                     for next_vert in self.vertices[current_vertex]:
-                        search_queue.enqueue(current_path.append(next_vert))
-        return -1  # target not found
+                        if next_vert is not None:
+                            current_path.append(next_vert)
+                            search_queue.enqueue(current_path)
+        return f"No route found from {starting_vertex} to {target}"
 
     def dfs(self, starting_vertex, target):  # Depth-First Search
         search_stack = Stack()
         search_stack.push([starting_vertex])
         visited = set()
-        paths_to_target = set()
+        paths_to_target = []
         while search_stack.size() > 0:
             current_path = search_stack.pop()
             current_vertex = current_path[-1]
             if current_vertex not in visited:
                 visited.add(current_vertex)
                 if current_vertex == target:
-                    paths_to_target.add(current_path)
+                    paths_to_target.append(current_path)
                 else:
                     for next_vert in self.vertices[current_vertex]:
-                        search_stack.push(current_path.append(next_vert))
+                        if next_vert is not None:
+                            current_path.append(next_vert)
+                            search_stack.push(current_path)
         if len(paths_to_target) > 0:
             return min(paths_to_target)
         else:
-            return -1  # target not found
+            # target not found
+            return f"No route found from {starting_vertex} to {target}"
```

Comment:
I like the way you are returning  with passed values back... easy to understand for user